kinto/api-test/test_kinto_heartbeat.py

```diff
@@ -19,11 +19,9 @@ def test_check_heartbeat(self):
         """Check heartbeat to make sure It's Alive."""
         resource = '__heartbeat__'
         response = self.client.get_request(resource, status_code=503)
-        self.assertIn('oauth', response)
         self.assertIn('cache', response)
         self.assertIn('storage', response)
         self.assertIn('permission', response)
-        self.assertEqual(response['oauth'], True)
         self.assertEqual(response['cache'], True)
         self.assertEqual(response['storage'], True)
         self.assertEqual(response['permission'], True)
```

Comment:
oauth is not 100% guaranteed to exist, so that's why I left it out. I
discovered this during older testing.

I can try with creating an expected response object though and see how
it behaves.


On Thu, Jan 14, 2016, at 01:14 PM, Peter deHaan wrote:
> In kinto/api-test/test_kinto_heartbeat.py[1]:


> >          self.assertIn('cache', response)
>          self.assertIn('storage', response)
>          self.assertIn('permission', response)
> -        self.assertEqual(response['oauth'], True)
>          self.assertEqual(response['cache'], True)
>          self.assertEqual(response['storage'], True)
>          self.assertEqual(response['permission'], True)
>          https://kinto.dev.mozaws.net/v1/__heartbeat__ shows a oauth
>          key. Not sure if there is an easier way to compare JSON
>          objects in Python. It may be easier to create an expected
>          response object and just compare it to the actual response,
>          since this test won't ever detect if new keys are added to
>          the response.


> In Node land, I'd use something like Joi[2] to create and validate a
> schema, but not sure what the best way in Python-land is.
> (https://github.com/monsur/jsoncompare looks promising -- or maybe i'm
> mentally over-optimizing something that needn't be optimized.)


> — Reply to this email directly or view it on GitHub[3].


>



Links:

  1. https://github.com/mozilla-services/services-test/pull/49#discussion_r49762097
  2. https://github.com/hapijs/joi
  3. https://github.com/mozilla-services/services-test/pull/49/files#r49762097